l1-contracts/src/core/Rollup.sol

```diff
@@ -89,8 +93,10 @@ contract Rollup is IRollup {
       revert Errors.Rollup__InvalidInHash(inHash, header.contentCommitment.inHash);
     }
 
-    IOutbox outbox = REGISTRY.getOutbox();
-    outbox.sendL1Messages(l2ToL1Msgs);
+    uint256 l2ToL1TreeHeight = MerkleLib.calculateTreeHeightFromSize(l2ToL1Msgs.length);
```

Comment:
You don't need to compute the height like this, you should be able to pick up information from the `contentCommitment.txTreeHeight` and then handle then umber of messages per tx.

Reply:
You will NOT have the `l2ToL1Msgs`, the entire `MessagesDecoder` can be purged when this pr is in.

Reply:
I closed that issue, at is was a duplicate of #5073.